src/rgw/rgw_rados.cc

```diff
@@ -6129,7 +6129,7 @@ int RGWRados::swift_versioning_copy(RGWObjectCtx& obj_ctx,
   r = get_bucket_info(obj_ctx, bucket_info.bucket.tenant, bucket_info.swift_ver_location, dest_bucket_info, NULL, NULL);
   if (r < 0) {
     ldout(cct, 10) << "failed to read dest bucket info: r=" << r << dendl;
-    return r;
+    return -ERR_PRECONDITION_FAILED;
```

Comment:
I think we should be more selective here. This translates to overwriting only those errors we expect like `ENOENT`.